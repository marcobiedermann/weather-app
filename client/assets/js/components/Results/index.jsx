import PropTypes from 'prop-types';
import React from 'react';
import Result from '../../containers/Result';

const Results = props => {
  const { results, ...otherProps } = props;

  return (
    <ul {...otherProps}>
      {results.map(result => (
        <li key={result.id}>
          <Result {...result} />
        </li>
      ))}
    </ul>
  );
};

Results.propTypes = {
  results: PropTypes.arrayOf(PropTypes.shape()),
};

Results.defaultProps = {
  results: [],
};

export default Results;
