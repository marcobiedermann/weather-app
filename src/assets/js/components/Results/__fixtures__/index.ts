import resultFixtures from '../../Result/__fixtures__';

const results = [resultFixtures];

export default results;
