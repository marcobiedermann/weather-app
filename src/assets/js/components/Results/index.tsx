import React, { FC } from 'react';
import Result, { ResultProps } from '../Result';
import styles from './style.module.css';

export interface ResultsProps {
  results: ResultProps[];
}

const Results: FC<ResultsProps> = (props) => {
  const { results } = props;

  return (
    <ul className={styles.results}>
      {results.map((result) => (
        <li key={result.id}>
          <Result {...result} />
        </li>
      ))}
    </ul>
  );
};

export default Results;
