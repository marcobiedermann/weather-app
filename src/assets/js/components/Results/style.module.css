.results {
  display: grid;
  gap: 0.5em;
  list-style: none;
  margin: 0;
  padding: 0;
}
