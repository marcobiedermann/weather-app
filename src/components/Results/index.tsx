import Result, { ResultProps } from '../Result';
import styles from './style.module.css';

interface ResultsProps {
  results: ResultProps[];
}

function Results(props: ResultsProps): JSX.Element {
  const { results } = props;

  return (
    <ul className={styles.results}>
      {results.map((result) => (
        <li key={result.id}>
          <Result {...result} />
        </li>
      ))}
    </ul>
  );
}

export type { ResultsProps };
export default Results;
