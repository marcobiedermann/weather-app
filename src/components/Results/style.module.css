.results {
  display: grid;
  list-style: none;
  margin: 0;
  padding: 0;
}

.results li + li {
  border-top: 1px solid #fcfcfc;
  padding-block: 0.5em;
}
