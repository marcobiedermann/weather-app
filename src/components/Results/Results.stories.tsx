import type { Meta, StoryObj } from '@storybook/react';
import Results from '.';

const meta: Meta<typeof Results> = {
  component: Results,
  args: {
    results: [
      {
        id: 2950159,
        name: 'Berlin',
        main: {
          temp: 18.54,
        },
        sys: {
          country: 'DE',
        },
      },
    ],
  },
};

type Story = StoryObj<typeof Results>;

const Default: Story = {};

export { Default };
export default meta;
