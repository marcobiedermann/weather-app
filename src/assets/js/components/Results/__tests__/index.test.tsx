import { render } from '@testing-library/react';
import React from 'react';
import Results from '..';
import resultsFixtures from '../__fixtures__';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));

describe('components/Results', () => {
  it('renders correctly', () => {
    expect.assertions(1);

    const { asFragment } = render(<Results results={resultsFixtures} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
