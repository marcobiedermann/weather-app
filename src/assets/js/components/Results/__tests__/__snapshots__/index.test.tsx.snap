// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Results renders correctly 1`] = `
<DocumentFragment>
  <ul
    class="results"
  >
    <li>
      <div
        class="result"
      >
        <h3>
          Berlin, DE
        </h3>
        12.61
        <button
          type="button"
        >
          translation:add
        </button>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
